datasets/opus_dogc/opus_dogc.py

```diff
@@ -0,0 +1,108 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""OPUS DOGC dataset."""
+
+import xml.etree.ElementTree as ET
+
+import datasets
+
+
+_DESCRIPTION = """\
+This is a collection of documents from the Official Journal of the Government of Catalonia, in Catalan and Spanish \
+languages, provided by Antoni Oliver Gonzalez from the Universitat Oberta de Catalunya.
+"""
+
+_CITATION = """\
+@inproceedings{tiedemann-2012-parallel,
+    title = "Parallel Data, Tools and Interfaces in {OPUS}",
+    author = {Tiedemann, J{\"o}rg},
+    booktitle = "Proceedings of the Eighth International Conference on Language Resources and Evaluation ({LREC}'12)",
+    month = may,
+    year = "2012",
+    address = "Istanbul, Turkey",
+    publisher = "European Language Resources Association (ELRA)",
+    url = "http://www.lrec-conf.org/proceedings/lrec2012/pdf/463_Paper.pdf",
+    pages = "2214--2218",
+    abstract = "This paper presents the current status of OPUS, a growing language resource of parallel corpora and related tools. The focus in OPUS is to provide freely available data sets in various formats together with basic annotation to be useful for applications in computational linguistics, translation studies and cross-linguistic corpus studies. In this paper, we report about new data sets and their features, additional annotation tools and models provided from the website and essential interfaces and on-line services included in the project.",
+}
+"""
+
+_URL = "http://opus.nlpl.eu/DOGC.php"
+_FILE_FORMATS = ["tmx"]
+_URLS = {"tmx": "http://opus.nlpl.eu/download.php?f=DOGC/v2/tmx/ca-es.tmx.gz"}
+
+
+class OpusDogcConfig(datasets.BuilderConfig):
+    """ BuilderConfig for OpusDogcConfig."""
+
+    def __init__(self, file_format=None, **kwargs):
+        """
+
+        Args:
+            file_format: language of the subdataset.
+            **kwargs: keyword arguments forwarded to super.
+        """
+        super().__init__(
+            name=file_format, description=f"OPUS DOGC dataset from source file format {file_format}.", **kwargs
+        )
+        self.file_format = file_format
+
+
+class OpusDogc(datasets.GeneratorBasedBuilder):
+    """OPUS DOGC dataset."""
+
+    BUILDER_CONFIG_CLASS = OpusDogcConfig
+    BUILDER_CONFIGS = [OpusDogcConfig(file_format=file_format) for file_format in _FILE_FORMATS]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "ca": datasets.Value("string"),
+                    "es": datasets.Value("string"),
```

Comment:
Could you use the `Translation` feature type ?
see [flores](https://github.com/huggingface/datasets/blob/master/datasets/flores/flores.py) for an example of usage

Reply:
Done, @lhoestq